### PR TITLE
Heart system fix for compatibility with possession mechanic

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
There will only be one HeartSystem in the current scene that's a component of the newly created LifeBar prefab.

This HeartSystem's `AssociateWith(Player player)` function can be called to associate it with a new Player object at runtime.

This should allow us to move forward with the possession mechanic and script reassociation.

### CODE REVIEW

File: Assets/Scripts/Player.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 public class Player : Entity
 {
     public delegate void HealthChanged();
-    public static event HealthChanged healthChangedEvent;
+    public event HealthChanged healthChangedEvent;
```
Comment (Jazy5552) [inline Assets/Scripts/Player.cs]: :)

**Comment by AleksandrHovhannisyan** [comment]
I mainly made these changes because Ed requested them, and I think it makes more sense this way. The problem is that upon possession, if the enemy has a different sprite and shape, we should transfer all those characteristics to the player. Instead of copying all the components of the Enemy and using them to override the Player's, it'll be easier to just switch who we consider the "player" to the enemy.

**Comment by klein1992** [comment]
The only issue I find with Jazy's idea is that the GameObject parent could potentially have many children meshes. And am not clear how much more beneficial Jazy's idea is or where improvement lies.

**Comment by Jazy5552** [comment]
Well no, my idea would mean that instead of deleting the player object, we just delete the player's physical objects (Like a gameobject called `model`), and replace it with the enemies. Like if in my prefab you replaced the cube and sphere with a cylinder or something.

What I am trying to avoid is de-referencing the player GameObject by deleting it. That would mean that everyone that has a reference to it would have to constantly check to see if it is valid, and get the player by it's tag again (Which is an expensive operation but im not too concerned about optimization).

**Comment by AleksandrHovhannisyan** [comment]
Hmm, that's actually a good point, Jazy. In that case, if you want, we can ditch this PR and think of an alternative.

**Pull request merged**